### ISSUE TITLE
[Request] Fix and enhance several issues with the Fleet/Agent proxy doc

### ISSUE THREAD

**Issue by joshdover** [elastic/observability-docs#2160]
## Description

There's several issues/inaccuracies in the [current documentation for Fleet & Agent's support of proxies](https://www.elastic.co/guide/en/fleet/current/fleet-agent-proxy-support.html) as well as some additions that need to be made. For reference, what we support is laid out in [this doc](https://docs.google.com/document/d/1MadI7N4fBjT67k3kM7lYZQJXKSZtUlaU-xYbkR8WL78/edit#bookmark=id.uc1lbhza5tgw).

### Clarification on support for proxy in Fleet-managed Agent

The docs has some inaccuracies and confusing headings.

Under the "Set Elastic Agent proxy settings in a standalone agent policy" header it says "Currently, we only offer a way to modify these for standalone agents. The ability to set these for Fleet-managed agents is [under consideration](https://github.com/elastic/kibana/issues/140533)." It then goes on to show how we action _do support proxies for Fleet-managed agents_, including a section "set the proxy for communicating with ES" where we show a screenshot of showing this in the UI. I think we should reconfigure this page a bit to make this distinction clear.

Suggested edits:
- Create two separate sections, one for standalone and one for managed Agents
- In the managed agent section, include a section on how to configure the ES output proxy and the [existing section](https://www.elastic.co/guide/en/fleet/current/fleet-agent-proxy-support.html#cli-proxy-settings) on how to configure the proxy for communicating with Fleet Server
- In the standalone agent section, include only how to configure the ES output proxy
- Remove the language about not supporting proxies for Fleet-managed agents

### Add details on how to configure certificates for HTTPS proxies

We have support for setting a custom CA, certificate, and key for trusting certs from a proxy, however this is not currently included in the documentation. IMO a dedicated section separate from the managed and standalone sections for this would be helpful, with links from those sections to this one.

For the output proxy to ES, essentially the user just needs to add the proxy's cert/key/ca to the Elasticsearch output's regular cert/key/ca settings, alongside any required to trust Elasticsearch's cert as well. These are [documented in detail in Beats](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html#ssl-client-config), but we should add some basic examples to this doc too.

For the proxy for communicating with Fleet, we have limited to support for the necessary configs on the CLI at enroll time via the `--certificate-authorities` option. One caveat here is that the CA has to already be trusted by the OS and there's not a way to specify a cert and key directly. We do however also support a `--ca-sha256` option to trust a CA based on it's hash pin.

### Resulting outline

- When to configure proxy settings
- Set proxy settings in host environment variables
- Set proxy settings for Fleet-managed Elastic Agents
	- For communicating with Elasticsearch
	- For communicating with Fleet Server
- Set proxy settings for standalone Elastic Agents
	- For communicating with Elasticsearch
- Configure trusted certificates with HTTPS proxies
	- For communicating with Elasticsearch
	- For communicating with Fleet Server
- Set the proxy URL of the Elastic Package Registry

## Collaboration

**Contact Person:** @joshdover
Informed: @nimarezainia 

## Suggested Target Release

Ready to go now, already applies. Future proxy enhancements scheduled for 8.6 which are separate from these updates.

## Stakeholders

@rstatsinger

**Comment by nimarezainia** [comment]
Lets address this after https://github.com/elastic/kibana/issues/140533 is completed - so that the proxy documentation is updated with the latest.

**Comment by nimarezainia** [comment]
@dedemorton this needs to be moved to 8.7. I'm preparing the document for it but we hit a bug that is targeted for 8.7

**Comment by dedemorton** [comment]
@nimarezainia I've added it as an 8.7-request. Would you consider opening a PR instead of adding your content to a google doc? The problem with using a google doc is that the review comments get lost over time and aren't reflected in the PR (so no history). Also with a google doc it's sometimes more difficult to understand where you want the content additions/changes to go.

**Comment by nimarezainia** [comment]
> 

@dedemorton do you mean to convert what I have into asciidoc? I would prefer to carry on working on the google doc as I can readily add diagrams and get them reviewed by the team. Could we not finalize in the google docs and then convert everything into asciidoc in one shot?

**Comment by nimarezainia** [comment]
closing in favour of https://github.com/elastic/observability-docs/issues/2394

**Comment by dedemorton** [comment]
@nimarezainia 
>Could we not finalize in the google docs and then convert everything into asciidoc in one shot?

tl;dr It's perfectly OK to use google docs as a form of "knowledge transfer" to the writer who will work on the docs. You can explain the new feature, how it works, how it impacts the users, what the benefits are, and identify the tasks users need to perform (demos are good too)...but putting docs into a google document and thinking we can just convert them into asciidoc doesn't work very well for us.

There are a few problems with this approach:

1. Writers aren't involved in the discussion early enough, which means we can't ask questions early on that influence the direction of the document. Sometimes this means we end up inheriting documents that don't follow best technical writing practices and might not contain content that we, as professional writers, believe it should. Ironically rewriting a document can sometimes be _more_ work than writing it from scratch because we have to honor someone else's vision for the content. And it's especially difficult if we weren't involved in technical discussions and don't understand the technical underpinnings.
2. A google doc does not accurately capture how the  content fits in with our existing docs.  It's never a simple conversion into asciidoc. Docs need to follow technical writing best practices, including Elastic style; the procedures need to be consistent with the rest of our docs; and the content needs to be integrated with existing content. We can help you with _all_ of those things, but when you bypass our usual process (PRs), you miss the opportunity to learn from us, and as a result, we spend more time on cleanup.
3. Related to my original comment, the initial technical review feedback gets lost, which means we risk revisiting discussions that already happened in private before we were involved, and if we have questions later on, we can't search GitHub to understand the history.
4. I think it might also make it harder for us to get review involvement when we do put up the PR because developers have already given input. (Just speculating here because turnaround times for reviews on this project has been very disappointing.)

I would say...work in your google doc to get input on things like graphics, but when the content is ready for a technical review/signoff, that should happen in GitHub just like all the other "code" we write (remember..."docs as code" is our mantra).

I personally feel that it's also important for authors to get accurately credited for their work.

Does that make sense?

**Comment by nimarezainia** [comment]
Thanks @dedemorton. I don't mean to create extra work for the docs teams. I am happy to (and I think I have) involve the docs team early on to review the documents. I just create the google doc as an easier way to communicate the content and get the graphics and content out there.

Unfortunately I am unable to create PRs on this content as I am not that well versed with asciidoc for complex documentation.

adding @mukeshelastic for info as well.
